src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,959 @@
+        required: false,
+        visibleDependency: `property${index}`,
+        visiblePredicate: (dependentValue: Property) =>
+            dependentValue !== Property.None,
+    }
+
+    return propertyColor
+}
+
+class VisualizerGrid extends VisualizerDefault {
+    name = 'Grid'
+    description =
+        'This visualizer puts any number sequence in a spiral or in '
+        + 'rows and allows you to highlight numbers based on various'
+        + ' properties.'
+        + ' One use of this tool is to put the natural numbers in a spiral'
+        + " and to highlight the prime ones, which is known as Ulam's "
+        + 'spiral. '
+        + 'Another use of this tool is highlighting properties such as'
+        + ' whether a number is abundant, polygonal, '
+        + 'or in some other sequence.'
+
+    //Grid variables
+    amountOfNumbers = 40000
+    currentSequenceIndex = 0
+    startingIndex = 0
+    currentNumber = 0n
+    showNumbers = false
+    preset = Preset.Custom
+    pathType = PathType.Spiral
+    resetAndAugmentByOne = false
+    squareValues = false
+    backgroundColor = BLACK
+    numberColor = WHITE
+
+    //Path variables
+    x = 0
+    y = 0
+    scalingFactor = 0
+    currentDirection = Direction.Right
+    numberToTurnAtForSpiral = 0
+    incrementForNumberToTurnAt = 1
+    whetherIncrementShouldIncrement = true
+
+    //Properties
+    propertyObjects = [
+        {
+            property: Property.Prime,
+            visualization: PropertyVisualization.Primary_Color,
+            color: RED,
+        },
+    ]
+
+    params: {[key: string]: ParamInterface} = {
+        preset: {
+            value: this.preset,
+            from: Preset,
+            displayName: 'Presets',
+            required: false,
+            description:
+                'Note: If a preset is selected, properties no longer function.',
+        },
+        amountOfNumbers: {
+            value: this.amountOfNumbers,
+            displayName: 'Amount of Numbers',
+            required: false,
+            description:
+                'Note: There is an increased lag time over 10,000 numbers',
+        },
+        startingIndex: {
+            value: this.startingIndex,
+            displayName: 'Starting Index',
+            required: false,
+            description: 'Note: This does not accept negative numbers.',
+        },
+        pathType: {
+            value: this.pathType,
+            from: PathType,
+            displayName: 'How path moves about grid',
+            required: false,
+        },
+        showNumbers: {
+            value: this.showNumbers,
+            forceType: 'boolean',
+            displayName: 'Show numbers',
+            required: false,
+            description:
+                'Note: Show_Path and Show_Numbers default to 400 numbers',
+        },
+        numberColor: {
+            value: this.numberColor,
+            forceType: 'color',
+            displayName: 'Number color:',
```

Comment:
Kate: Since the colour dialogs pop up to the left of the displayName, remove the colon here and in 'Background color:'

See https://github.com/numberscope/frontscope/pull/164#discussion_r1046399515 for the original comment.

Reply:
Should be done in https://github.com/numberscope/frontscope/pull/230/commits/b8fdb093b701217c19d8a13a7886748bafdcf132. Someone please check and resolve this conversation if so.

Reply:
looks good